prov/rxd/src/rxd_ep.c

```diff
@@ -1252,8 +1252,9 @@ int rxd_endpoint(struct fid_domain *domain, struct fi_info *info,
 	rxd_ep->prefix_size = dg_info->ep_attr->msg_prefix_size;
 	fi_freeinfo(dg_info);
 
-	rxd_ep->rx_size = info->rx_attr->size;
-	rxd_ep->tx_size = info->tx_attr->size;
+	rxd_ep->rx_size = MIN(dg_info->rx_attr->size, info->rx_attr->size);
+	rxd_ep->tx_size = MIN(dg_info->tx_attr->size, info->tx_attr->size);
```

Comment:
Why is this needed?  The info sizes should already be adjusted based on the dgram size.  The app expects to use the info sizes itself.

Reply:
With the proposed changes, the code looks wrong.  The app is expecting to send/receive the info size values.  Those should already be set based on the underlying dg_info sizes. I don't understand what the failure actually is